### ISSUE TITLE
Regarding CA1033 "Implement IDisposable correctly"

### ISSUE THREAD

**Issue by Arlodotexe** [CommunityToolkit/Tooling-Windows-Submodule#63]
[Error CA1033](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1033) indicates the IDisposable isn't implemented correctly, and needs to be virtual for unsealed types.

This doesn't appear to have been used in the old repo and is blocking the porting of some controls. This needs clarification, so this error has been suppressed on some components for now, keeping the original code as-is.

@michael-hawker Do we want this to stay enabled going forward?

Documentation seems to suggest that this should never be suppressed, but conversations with @Sergio0694 and other members of the community indicate that this isn't critical.

_Originally posted by @Arlodotexe in https://github.com/CommunityToolkit/Windows/issues/72#issuecomment-1565086190_

**Comment by michael-hawker** [comment]
Makes sense that we should update the pattern and not ignore the warning.

I believe in most cases in the Toolkit we're just dealing with managed resources?

Probably also relates to better implementations and usages around weak event patterns?

**Comment by Sergio0694** [comment]
> "I believe in most cases in the Toolkit we're just dealing with managed resources?"

Yes. AFAIK there isn't a single case in the entire Toolkit (both Windows and .NET) with wrapped native resources.

**Comment by Arlodotexe** [comment]
Sounds like we should do this properly for derived types with native resources.
